ui/widgets/trending/src/components/cards/latest-topics.tsx

```diff
@@ -29,27 +31,203 @@ export type LatestTopicsProps = {
-  handleUnsubscribeTopic: (topic: string) => void;
+  showLoginModal: () => void;
+  refetchTagSubscriptions: () => void;
 };
 
 export const LatestTopics: React.FC<LatestTopicsProps> = props => {
   const {
     onClickTopic,
-    handleSubscribeTopic,
-    handleUnsubscribeTopic,
     titleLabel,
     tagSubtitleLabel,
     tags,
     isLoadingTags,
-    isProcessingTags,
     noTagsLabel,
     subscribeLabel,
     subscribedLabel,
     unsubscribeLabel,
-    subscribedTags,
+    receivedTags,
+    tagSubscriptionsId,
+    isLoggedIn,
+    showLoginModal,
+    refetchTagSubscriptions,
   } = props;
 
+  const [localSubscribedTags, setLocalSubscribedTags] = useState([]);
+  const subscriptionId = useRef(null);
+  const localTagsInitialized = useRef(null);
+  const timer = useRef(null);
+  const [tagsQueue, setTagsQueue] = useState([]);
+  const localSubscribedTagsRef = useRef([]);
+
+  //reset state when user logs out
+  useEffect(() => {
+    if (!isLoggedIn) {
+      setLocalSubscribedTags([]);
+      localSubscribedTagsRef.current = [];
+      localTagsInitialized.current = null;
+      subscriptionId.current = null;
+    }
+  }, [isLoggedIn]);
+
+  useEffect(() => {
+    if (receivedTags && !localTagsInitialized.current) {
+      setLocalSubscribedTags([...new Set(receivedTags)]);
+      localTagsInitialized.current = true;
+    }
+    //prevents reseting interests with []
+    if (receivedTags && tagsQueue.length === 0) {
+      localSubscribedTagsRef.current = receivedTags;
+    }
+  }, [receivedTags, localTagsInitialized, tagsQueue.length]);
+
+  useEffect(() => {
+    if (tagSubscriptionsId) {
+      subscriptionId.current = tagSubscriptionsId;
+    }
+  }, [tagSubscriptionsId]);
+
+  const sdk = getSDK();
+
+  const [createInterestsMutation, { loading, error }] = useCreateInterestsMutation({
+    context: { source: sdk.services.gql.contextSources.composeDB },
+  });
+
+  const [updateInterestsMutation, { loading: updateLoading, error: updateError }] =
+    useUpdateInterestsMutation({
+      context: { source: sdk.services.gql.contextSources.composeDB },
+    });
+
+  useEffect(() => {
+    if (!localTagsInitialized.current || receivedTags === null) return;
+    if (loading || updateLoading) return;
+
+    if (!isEqual(localSubscribedTagsRef.current, receivedTags)) {
+      if (subscriptionId.current) {
+        updateInterestsMutation({
+          variables: {
+            i: {
+              id: subscriptionId.current,
+              content: {
+                topics: [...new Set(localSubscribedTagsRef.current)].map(tag => ({
+                  value: tag,
+                  labelType: sdk.services.gql.labelTypes.INTEREST,
+                })),
+              },
+            },
+          },
+          optimisticResponse: {
+            updateAkashaProfileInterests: {
+              clientMutationId: null,
+              document: {
+                did: { id: 'did:pkh:eip155:5:0xe92bbe3e927f73106e57ed43fd3f2acf51035128' },
```

Comment:
why this specific DID?

Reply:
you could use something like 0x address `did:pkh:eip155:1:0x0000000000000000000000000000000000000000`

Reply:
or even better, fetch the DID from the currently signed in user `getCurrentUser`

Reply:
this should not be hardcoded like here

Reply:
thanks for spotting this.